make test has some tests failing with assert 1 == 5

## Describe the bug
```

_______________________________________________________ test_failure_not_trestle[trestle author docs setup -tn test] ________________________________________________________
[gw15] linux -- Python 3.9.9 /home/degenaro/venv.oscal-catalog-to-csv/bin/python

command_string = 'trestle author docs setup -tn test', tmp_path = PosixPath('/tmp/pytest-of-degenaro/pytest-41/popen-gw15/test_failure_not_trestle_trest0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f5767061160>

    @pytest.mark.parametrize(
        'command_string', [('trestle author docs setup -tn test'), ('trestle author folders setup -tn test')]
    )
    def test_failure_not_trestle(command_string, tmp_path: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
        """Test for failure based on not in trestle directory."""
        monkeypatch.setattr(sys, 'argv', command_string.split())
        with pytest.raises(SystemExit) as wrapped_error:
            trestle.cli.run()
            # FIXME: Needs to be changed once implemented.
        assert wrapped_error.type == SystemExit
>       assert wrapped_error.value.code == 5
E       assert 1 == 5
E        +  where 1 = SystemExit(1).code
E        +    where SystemExit(1) = <ExceptionInfo SystemExit(1) tblen=3>.value

tests/trestle/core/commands/author/command_test.py:59: AssertionError

```

## To Reproduce

1. clone compliance-trestle repo
2. create venv
3. source venv
4. make install
5. make develop
6. make test

## Expected behavior

The tests should not fail.

## Screenshots / Logs.

Possible cause is that some tests are not considering the corruption of the cwd.
Here is a fix that worked, by inserting the line: monkeypatch.chdir(tmp_path)

```

@pytest.mark.parametrize(
    'command_string', [('trestle author docs setup -tn test'), ('trestle author folders setup -tn test')]
)
def test_failure_not_trestle(command_string, tmp_path: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
    """Test for failure based on not in trestle directory."""
    monkeypatch.setattr(sys, 'argv', command_string.split())
    monkeypatch.chdir(tmp_path)
    with pytest.raises(SystemExit) as wrapped_error:
        trestle.cli.run()
        # FIXME: Needs to be changed once implemented.
    assert wrapped_error.type == SystemExit
    assert wrapped_error.value.code == 5
```

Also, the hard coding of `5` should more reliably be replaced with `TRESTLE_ROOT_ERROR`.

## Environment

- OS: Red Hat Enterprise Linux release 8.7 (Ootpa)
- Python version: 3.9
- Installed packages: N/A

--- a/tests/trestle/core/commands/author/command_test.py
+++ b/tests/trestle/core/commands/author/command_test.py
@@ -22,6 +22,7 @@ from _pytest.monkeypatch import MonkeyPatch
 import pytest
 
 import trestle.cli
+from trestle.core.commands.common.return_codes import CmdReturnCodes
 
 
 def test_governed_docs_cli(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
@@ -32,7 +33,7 @@ def test_governed_docs_cli(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPat
         trestle.cli.run()
         # FIXME: Needs to be changed once implemented.
     assert wrapped_error.type == SystemExit
-    assert wrapped_error.value.code == 2
+    assert wrapped_error.value.code == CmdReturnCodes.INCORRECT_ARGS.value
 
 
 def test_governed_folders_cli(tmp_trestle_dir: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
@@ -43,7 +44,7 @@ def test_governed_folders_cli(tmp_trestle_dir: pathlib.Path, monkeypatch: Monkey
         trestle.cli.run()
         # FIXME: Needs to be changed once implemented.
     assert wrapped_error.type == SystemExit
-    assert wrapped_error.value.code == 2
+    assert wrapped_error.value.code == CmdReturnCodes.INCORRECT_ARGS.value
 
 
 @pytest.mark.parametrize(
@@ -52,8 +53,9 @@ def test_governed_folders_cli(tmp_trestle_dir: pathlib.Path, monkeypatch: Monkey
 def test_failure_not_trestle(command_string, tmp_path: pathlib.Path, monkeypatch: MonkeyPatch) -> None:
     """Test for failure based on not in trestle directory."""
     monkeypatch.setattr(sys, 'argv', command_string.split())
+    monkeypatch.chdir(tmp_path)
     with pytest.raises(SystemExit) as wrapped_error:
         trestle.cli.run()
         # FIXME: Needs to be changed once implemented.
     assert wrapped_error.type == SystemExit
-    assert wrapped_error.value.code == 5
+    assert wrapped_error.value.code == CmdReturnCodes.TRESTLE_ROOT_ERROR.value

--- a/tests/trestle/core/commands/href_test.py
+++ b/tests/trestle/core/commands/href_test.py
@@ -24,6 +24,7 @@ from tests import test_utils
 
 from trestle.cli import Trestle
 from trestle.common.model_utils import ModelUtils
+from trestle.core.commands.common.return_codes import CmdReturnCodes
 from trestle.core.models.file_content_type import FileContentType
 from trestle.oscal import profile
 
@@ -70,19 +71,20 @@ def test_href_failures(
     tmp_path: pathlib.Path, keep_cwd: pathlib.Path, simplified_nist_profile: profile.Profile, monkeypatch: MonkeyPatch
 ) -> None:
     """Test href failure modes."""
+    cmd_string = 'trestle href -n my_test_model -hr foobar'
+
+    # not in trestle project so fail
+    monkeypatch.setattr(sys, 'argv', cmd_string.split())
+    monkeypatch.chdir(tmp_path)
+    rc = Trestle().run()
+    assert rc == CmdReturnCodes.TRESTLE_ROOT_ERROR.value
+
     # prepare trestle project dir with the file
     models_path, profile_path = test_utils.prepare_trestle_project_dir(
         tmp_path,
         FileContentType.JSON,
         simplified_nist_profile,
         test_utils.PROFILES_DIR)
-
-    cmd_string = 'trestle href -n my_test_model -hr foobar'
-
-    # not in trestle project so fail
-    monkeypatch.setattr(sys, 'argv', cmd_string.split())
-    rc = Trestle().run()
-    assert rc == 5
 
     os.chdir(models_path)
 
@@ -93,4 +95,4 @@ def test_href_failures(
     simplified_nist_profile.oscal_write(profile_path)
     monkeypatch.setattr(sys, 'argv', cmd_string.split())
     rc = Trestle().run()
-    assert rc == 1
+    assert rc == CmdReturnCodes.COMMAND_ERROR.value

--- a/tests/trestle/core/commands/remove_test.py
+++ b/tests/trestle/core/commands/remove_test.py
@@ -27,6 +27,7 @@ from tests import test_utils
 
 import trestle.common.err as err
 from trestle.cli import Trestle
+from trestle.core.commands.common.return_codes import CmdReturnCodes
 from trestle.core.commands.remove import RemoveCmd
 from trestle.core.models.actions import RemoveAction
 from trestle.core.models.elements import Element, ElementPath
@@ -124,20 +125,20 @@ def test_remove_failure(tmp_path: pathlib.Path):
 def test_run_failure_switches(tmp_path: pathlib.Path, monkeypatch: MonkeyPatch):
     """Test failure of _run on bad switches for RemoveCmd."""
     # 1. Missing --file argument.
-    testargs = ['trestle', 'remove', '-e', 'catalog.metadata.roles']
+    testargs = ['trestle', 'remove', '-e', 'catalog.metadata.roles', '--trestle-root', str(tmp_path)]
     monkeypatch.setattr(sys, 'argv', testargs)
     with pytest.raises(SystemExit) as e:
         Trestle().run()
     assert e.type == SystemExit
-    assert e.value.code == 2
+    assert e.value.code == CmdReturnCodes.INCORRECT_ARGS.value
 
     # 2. Missing --element argument.
-    testargs = ['trestle', 'remove', '-f', './catalog.json']
+    testargs = ['trestle', 'remove', '-f', './catalog.json', '--trestle-root', str(tmp_path)]
     monkeypatch.setattr(sys, 'argv', testargs)
     with pytest.raises(SystemExit) as e:
         Trestle().run()
     assert e.type == SystemExit
-    assert e.value.code == 2
+    assert e.value.code == CmdReturnCodes.INCORRECT_ARGS.value
 
 
 def test_run_failure_nonexistent_element(
@@ -154,18 +155,27 @@ def test_run_failure_nonexistent_element(
     )
 
     # 1. self.remove() fails -- Should happen if wildcard is given, or nonexistent element.
-    testargs = ['trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.blah']
+    testargs = ['trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.blah', '--trestle-root', str(tmp_path)]
     monkeypatch.setattr(sys, 'argv', testargs)
     exitcode = Trestle().run()
-    assert exitcode == 5
+    assert exitcode == CmdReturnCodes.COMMAND_ERROR.value
 
     # 2. Corrupt json file
     source_file_path = pathlib.Path.joinpath(test_utils.JSON_TEST_DATA_PATH, 'bad_simple.json')
     shutil.copyfile(source_file_path, catalog_def_file)
-    testargs = ['trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.metadata.roles']
+    testargs = [
+        'trestle',
+        'remove',
+        '-f',
+        str(catalog_def_file),
+        '-e',
+        'catalog.metadata.roles',
+        '--trestle-root',
+        str(tmp_path)
+    ]
     monkeypatch.setattr(sys, 'argv', testargs)
     exitcode = Trestle().run()
-    assert exitcode == 5
+    assert exitcode == CmdReturnCodes.COMMAND_ERROR.value
 
 
 def test_run_failure_wildcard(tmp_path: pathlib.Path, sample_catalog_minimal: Catalog, monkeypatch: MonkeyPatch):
@@ -178,10 +188,10 @@ def test_run_failure_wildcard(tmp_path: pathlib.Path, sample_catalog_minimal: Ca
         sample_catalog_minimal,
         test_utils.CATALOGS_DIR
     )
-    testargs = ['trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.*']
+    testargs = ['trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.*', '--trestle-root', str(tmp_path)]
     monkeypatch.setattr(sys, 'argv', testargs)
     exitcode = Trestle().run()
-    assert exitcode == 5
+    assert exitcode == CmdReturnCodes.COMMAND_ERROR.value
 
 
 def test_run_failure_required_element(
@@ -198,10 +208,12 @@ def test_run_failure_required_element(
     )
     # 4. simulate() fails -- Should happen if required element is target for deletion
     monkeypatch.chdir(tmp_path)
-    testargs = ['trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.metadata']
+    testargs = [
+        'trestle', 'remove', '-f', str(catalog_def_file), '-e', 'catalog.metadata', '--trestle-root', str(tmp_path)
+    ]
     monkeypatch.setattr(sys, 'argv', testargs)
     exitcode = Trestle().run()
-    assert exitcode == 1
+    assert exitcode == CmdReturnCodes.COMMAND_ERROR.value
 
 
 def test_run_failure_project_not_found(
@@ -217,10 +229,10 @@ def test_run_failure_project_not_found(
         test_utils.CATALOGS_DIR
     )
     # 5. get_contextual_model_type() fails, i.e., "Trestle project not found"
-    testargs = ['trestle', 'remove', '-f', '/dev/null', '-e', 'catalog.metadata']
+    testargs = ['trestle', 'remove', '-f', '/dev/null', '-e', 'catalog.metadata', '--trestle-root', str(tmp_path)]
     monkeypatch.setattr(sys, 'argv', testargs)
     exitcode = Trestle().run()
-    assert exitcode == 5
+    assert exitcode == CmdReturnCodes.COMMAND_ERROR.value
 
 
 def test_run_failure_filenotfounderror(
@@ -238,11 +250,18 @@ def test_run_failure_filenotfounderror(
     # 6. oscal_read fails because file is not found
     # Must specify catalogs/ location, not catalogs/my_test_model/.
     testargs = [
-        'trestle', 'remove', '-f', re.sub('my_test_model/', '', str(catalog_def_file)), '-e', 'catalog.metadata'
+        'trestle',
+        'remove',
+        '-f',
+        re.sub('my_test_model/', '', str(catalog_def_file)),
+        '-e',
+        'catalog.metadata',
+        '--trestle-root',
+        str(tmp_path)
     ]
     monkeypatch.setattr(sys, 'argv', testargs)
     exitcode = Trestle().run()
-    assert exitcode == 5
+    assert exitcode == CmdReturnCodes.COMMAND_ERROR.value
 
 
 def test_run_failure_plan_execute(

